airgun/entities/hostcollection.py

```diff
@@ -31,12 +35,58 @@ def read(self, entity_name):
 
     def update(self, entity_name, values):
         view = self.navigate_to(self, 'Edit', entity_name=entity_name)
-        return view.fill(values)
+        return view.details.fill(values)
 
     def associate_host(self, entity_name, host_name):
         view = self.navigate_to(self, 'Edit', entity_name=entity_name)
         view.hosts.resources.add(host_name)
 
+    def manage_packages(
+            self, entity_name, content_type='Package', packages=None,
+            action='install', action_via='via Katello Agent'):
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        view.details.manage_packages.click()
+        view = HostCollectionManagePackagesView(view.browser)
+        if content_type is not None:
+            view.content_type.fill(content_type)
+        if packages is not None:
+            view.packages.fill(packages)
+        view.apply_action(action, action_via=action_via)
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+
+    def install_errata(self, entity_name, errata_id,
+                       install_via='via Katello agent'):
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        view.details.install_errata.click()
+        view = HostCollectionInstallErrataView(view.browser)
+        view.search.fill(errata_id)
+        view.table.row(Id=errata_id)[0].widget.fill(True)
+        if install_via:
```

Comment:
But you've already specified `install_via='via Katello agent'` by default, so this `if` doesn't make much sense. Maybe `install_via` should default to `None` instead, so by default we just click the button and only if `install_via` is specified we open actions list? Same applies to other helpers

Reply:
Here we have a dropdown you can select the via or click on the main button but if you click on the main button we have a dialog displayed, but if you select from the list there is no dialog. that way if a user want to click on the main button he has simply to not provide the install_via make it False of None and the install will be made by the default installation via mean eg clicking on the main button and clicking on confirm dialog.

Reply:
Makes sense, then we need this check. This reminds me `ActionsDropdown` widget btw. Have you checked it?
And what about other part of suggestion - having `install_via` set to `None` by default?

Reply:
effectively we are using ActionsDropdown so will change that with fill the action action and check if the dialog is displayed than confirm.